rfd/0005/README.mdx

```diff
@@ -0,0 +1,453 @@
+---
+authors: Nicolas Joseph <nicolas@fiveonefour.com>
+title: Moose CLI Version Management
+state: prediscussion
+---
+
+## Objective
+
+We propose a new version management system for the Moose CLI that ensures compatibility between the CLI and its associated
+language-specific libraries (moose-libs). The goal is to create a transparent version management experience where the correct 
+CLI version is automatically used based on project requirements, eliminating version mismatch issues between the CLI and libraries.
+
+## Scope
+
+This RFD focuses specifically on the Moose CLI version management system. The following are explicitly out of scope:
+
+- Language-specific libraries (moose-libs):
+  - These are managed through their respective language package managers (npm, pip, etc.)
+  - Installation and versioning handled by the project's dependency management system
+  - Libraries are project-local and not directly called by users
+  - Version compatibility is enforced through dependency specifications in package.json/requirements.txt
```

Comment:
Would this change how we develop moose-lib locally? I typically use `npm link`. When we `npm install`, I think the version is the actual version (i.e. 0.4.x), but the ts-moose-lib package.json has `0.0.2`. I'm not sure if it will get confused.

Reply:
that's a good point, I am thinking of adding a way to shortcut as an option. 

As I'm writing this, I was also thinking that since this is just a wrapper script that's going to be on the moose name, then you can still start moose in dev mode with directly accessing the binary.